repos/fdbt-site/src/pages/api/deletePassenger.ts

```diff
@@ -2,35 +2,40 @@ import { NextApiResponse } from 'next';
 import { deletePassengerTypeByNocCodeAndId, getGroupPassengerTypeDbsFromGlobalSettings } from '../../data/auroradb';
 import { redirectToError, redirectTo, getAndValidateNoc } from '../../utils/apiUtils/index';
 import { NextApiRequestWithSession } from '../../interfaces';
+import { globalSettingsDeleteEnabled } from '../../../src/constants/featureFlag';
 
 export default async (req: NextApiRequestWithSession, res: NextApiResponse): Promise<void> => {
-    try {
-        const { query } = req;
+    if (globalSettingsDeleteEnabled) {
+        try {
+            const { query } = req;
 
-        const id = Number(query?.id);
+            const id = Number(query?.id);
 
-        const isGroup = query?.isGroup;
+            const isGroup = query?.isGroup;
 
-        if (!Number.isInteger(id) || typeof isGroup !== 'string') {
-            throw new Error('insufficient data provided for delete query');
-        }
+            if (!Number.isInteger(id) || typeof isGroup !== 'string') {
+                throw new Error('insufficient data provided for delete query');
+            }
 
-        const nationalOperatorCode = getAndValidateNoc(req, res);
-        const groups = await getGroupPassengerTypeDbsFromGlobalSettings(nationalOperatorCode);
-        const groupsInUse = groups.some((group) =>
-            group.groupPassengerType.companions.some((individual) => individual.id === id),
-        );
+            const nationalOperatorCode = getAndValidateNoc(req, res);
+            const groups = await getGroupPassengerTypeDbsFromGlobalSettings(nationalOperatorCode);
+            const groupsInUse = groups.some((group) =>
+                group.groupPassengerType.companions.some((individual) => individual.id === id),
+            );
 
-        if (groupsInUse) {
-            throw new Error('This individual cannot be deleted because it is in use in a group.');
-        }
+            if (groupsInUse) {
+                throw new Error('This individual cannot be deleted because it is in use in a group.');
+            }
 
-        await deletePassengerTypeByNocCodeAndId(id, nationalOperatorCode, isGroup === 'true');
+            await deletePassengerTypeByNocCodeAndId(id, nationalOperatorCode, isGroup === 'true');
 
-        redirectTo(res, '/viewPassengerTypes');
-    } catch (error) {
-        const message = 'There was a problem deleting the selected passenger or group';
+            redirectTo(res, '/viewPassengerTypes');
+        } catch (error) {
+            const message = 'There was a problem deleting the selected passenger or group';
 
-        redirectToError(res, message, 'api.deletePassenger', error);
+            redirectToError(res, message, 'api.deletePassenger', error);
+        }
+    } else {
+        redirectTo(res, '/error');
```

Comment:
FYI there's a helper for this `redirectToError` that handles logging it and stuff. Althernatively you could just throw an error inside the `try` if it's not enabled? Either way 👍